core/src/audio_subsystem.cpp

```diff
@@ -0,0 +1,206 @@
+#include <knoting/audio_subsystem.h>
+
+#include <knoting/audio_source.h>
+
+namespace knot {
+
+AudioSubsystem::AudioSubsystem() = default;
+
+void AudioSubsystem::on_awake() {
+    void* extra_driver_data = nullptr;
+    m_result = System_Create(&m_system);
+    ErrorCheck(m_result);
+    m_result = m_system->init(512, FMOD_INIT_NORMAL, extra_driver_data);
+    ErrorCheck(m_result);
+    m_result = m_system->set3DSettings(1.0f, 1.0f, 1.0f);
+    ErrorCheck(m_result);
+    log::debug("AudioSubsystem created!");
+}
+
+void AudioSubsystem::on_destroy() {
+    m_result = m_system->close();
+    ErrorCheck(m_result);
+    m_result = m_system->release();
+    ErrorCheck(m_result);
+}
+
+void AudioSubsystem::play(components::AudioSource& source) {
+    source.set_playing(false);
```

Comment:
We shouldn't need to do this, unless the m_isPlaying variable, is not being set in the constructor